### PR TITLE
call gutentags#get_project_root with absolute path to current file

### PR DESCRIPTION
**Pull request by dnhgff**
this is my proposed solution to #21

### CODE REVIEW

File: autoload/gutentags.vim [hunk 1]
```diff
@@ -98,7 +98,7 @@ function! gutentags#setup_gutentags() abort
     " Try and find what tags file we should manage.
     call gutentags#trace("Scanning buffer '" . bufname('%') . "' for gutentags setup...")
     try
-        let b:gutentags_root = gutentags#get_project_root(expand('%:h'))
+        let b:gutentags_root = gutentags#get_project_root(expand('%:p:h'))
```
Comment (justinmk) [inline autoload/gutentags.vim]: `expand()` should always be called with `1` as the second argument, to avoid `'wildignore'` pain.

Comment (dnhgff) [inline autoload/gutentags.vim]: thanks, commit added (after i :h expand for own learning :P)

Comment (ludovicchabant) [inline autoload/gutentags.vim]: Interesting, I didn't know about passing `1` to `expand()`!

**Comment by ludovicchabant** [comment]
Thanks for that patch!

File: autoload/gutentags.vim (hunk 1 above)
Comment (justinmk) [inline autoload/gutentags.vim]: There's several other expand() calls in the plugin you may want to look at.

Comment (ludovicchabant) [inline autoload/gutentags.vim]: Actually it looks like it's OK in those cases. Calling `expand()` with an argument that starts with any of the special stuff (`%`, `#`, `<sfile>`, etc) makes it somehow not use `wildignore`. At least that's what I'm seeing here when I actually open one of my ignored file (like a `.pyc` file).

Comment (justinmk) [inline autoload/gutentags.vim]: I suggest being defensive about it.

**Pull request merged**